java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/db/DBEntityFromTables.java

```diff
@@ -0,0 +1,139 @@
+        if (!COMMAND_ENTITY_FROM_TABLES.equals(command)) {
+            return null;
+        }
+        Lookup.getDefault().lookup(LspServerState.class).openedProjects().thenAccept((projects) -> {
+            if (projects.length > 0) {
+                createEntityClassesInProject(client, projects[0]);
+            } else {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoProject()));
+            }
+
+        });
+        return null;
+    }
+
+    private CompletableFuture createEntityClassesInProject(NbCodeLanguageClient client, Project prj) {
+        try {
+            DatabaseConnection connection = ConnectionManager.getDefault().getPreferredConnection(true);
+            if (connection == null) {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoDbConn()));
+                return null;
+            }
+            ConnectionManager.getDefault().connect(connection);
+            SourceGroup[] sr = ProjectUtils.getSources(prj).getSourceGroups(JavaProjectConstants.SOURCES_TYPE_JAVA);
+            if (sr.length < 1) {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoSourceRoot()));
+                return null;
+            }
+            Connection conn = connection.getJDBCConnection();
+            ResultSet rs = conn.getMetaData().getTables(conn.getCatalog(), conn.getSchema(), "%", new String[]{"TABLE", "VIEW"}); //NOI18N
+            List<QuickPickItem> dbItems = new ArrayList<>();
+            while (rs.next()) {
+                dbItems.add(new QuickPickItem(rs.getString("TABLE_NAME"))); //NOI18N
+            }
+            return client.showQuickPick(new ShowQuickPickParams(Bundle.MSG_SelectTables(), true, dbItems))
+                    .thenApply(items -> items.stream().map(item -> item.getLabel()).collect(Collectors.toList()))
+                    .thenAccept(tables -> {
+                        client.showInputBox(new ShowInputBoxParams(Bundle.MSG_EnterPackageName(), "")) //NOI18N
+                                .thenAccept(packageName -> {
+                                    EntitiesFromDBGenerator generator = new EntitiesFromDBGenerator(tables, true, packageName, sr[0], connection, prj, null);
+                                    ProgressContributor pc = BasicAggregateProgressFactory.createProgressContributor("entity"); //NOI18N
+                                    try {
+                                        generator.generate(pc);
+                                    } catch (SQLException | IOException ex) {
+                                        client.showMessage(new MessageParams(MessageType.Error, ex.getMessage()));
+                                    }
+                                });
+                        return;
+                    });
+        } catch (SQLException | IllegalArgumentException | DatabaseException ex) {
+            System.out.println(ex.getMessage());
```

Comment:
make this ` client.showMessage` instead, or use `NotifyDescriptor` + `DialogDisplayer.getDefault().notify()` that will be automagically remoted to the client.